### PR TITLE
dartlang starter kit

### PR DESCRIPTION
**Pull request by jerold**
Allows the player to write their bot in Dart.

Installing Dart: https://www.dartlang.org/install/linux

### CODE REVIEW

File: airesources/Dart/lib/src/logger.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class Logger {
+  String _fileName;
+
+  Logger._();
+
+  void call(String log) {
+    if (_fileName == null) return;
+    new File(_fileName).writeAsStringSync("$log\n", mode: FileMode.WRITE_ONLY_APPEND);
```
Comment (Snaar) [inline airesources/Dart/lib/src/logger.dart]: Let's use WRITE_ONLY mode. Not overwriting log by default would be confusing. All other kits overwrite.

File: airesources/Dart/lib/src/position.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    final double x = target.x + radius * cos(angleRad);
+    final double y = target.y + radius * sin(angleRad);
+    return new Position(x: x, y: y);
+  }
+}
+
+int toDegrees(double radians) => (radians * 180 / PI).round();
+
+int angleRadToDegClipped(double angleRad) {
+  return ((toDegrees(angleRad) % 360) + 360) % 360;
```
Comment (Snaar) [inline airesources/Dart/lib/src/position.dart]: Dart's mod operator always produces positive numbers. So `return toDegrees(angleRad) % 360;` should be sufficient.

**Comment by Snaar** [comment]
Ok this looks good, just say when you are done with updates you want to do for this PR.

**Comment by jerold** [comment]
Should be good to go. Thanks @Snaar for the review!

**Comment by jerold** [comment]
All done with updates. 😀

**Pull request merged**